drivers/storage/portworx/util/util.go

```diff
@@ -638,17 +630,32 @@ func AppendTLSEnv(clusterSpec *corev1.StorageClusterSpec, envMap map[string]*v1.
 		//      value: <default>
 		//    - name: PX_ENABLE_TLS
 		//	    value: "true"
-		if clusterSpec.Security.TLS.AdvancedTLSOptions != nil &&
-			!IsEmptyOrNilCertLocation(clusterSpec.Security.TLS.AdvancedTLSOptions.RootCA) {
+		if clusterSpec.Security.TLS != nil && !IsEmptyOrNilCertLocation(clusterSpec.Security.TLS.RootCA) {
 			// if no Root CA is specified, we assume server certs are signed by a well known cert authority. Do not need a CA
-			logrus.Infof("Secret name containing CA cert: %v", DefaultCASecretName)
-			envMap[EnvKeyCASecretName] = &v1.EnvVar{
-				Name:  EnvKeyCASecretName,
-				Value: DefaultCASecretName,
-			}
-			envMap[EnvKeyCASecretKey] = &v1.EnvVar{
-				Name:  EnvKeyCASecretKey,
-				Value: DefaultCASecretKey,
+
+			// if the user specified a CA cert though k8s secret in spec.tls.rootCA.secretRef, use that
+			if !IsEmptyOrNilSecretReference(clusterSpec.Security.TLS.RootCA.SecretRef) {
+				rootCASecret := clusterSpec.Security.TLS.RootCA.SecretRef
+				logrus.Infof("Secret name containing CA cert: %v", DefaultCASecretName)
+				envMap[EnvKeyCASecretName] = &v1.EnvVar{
+					Name:  EnvKeyCASecretName,
+					Value: rootCASecret.SecretName,
+				}
+				envMap[EnvKeyCASecretKey] = &v1.EnvVar{
+					Name:  EnvKeyCASecretKey,
+					Value: rootCASecret.SecretKey,
+				}
+			} else {
+				// The user installed a CA cert on the host file system. We assume that user has also uploaded it to the default k8s secret with the default key
+				logrus.Infof("Secret name containing CA cert: %v", DefaultCASecretName)
+				envMap[EnvKeyCASecretName] = &v1.EnvVar{
+					Name:  EnvKeyCASecretName,
+					Value: DefaultCASecretName,
+				}
+				envMap[EnvKeyCASecretKey] = &v1.EnvVar{
+					Name:  EnvKeyCASecretKey,
+					Value: DefaultCASecretKey,
+				}
```

Comment:
Sorry to go back and forth on this. As you mentioned the certs could be signed by a known CA in which case the clients don't need the CA cert. Which also means user won't create a secret with any CA cert. What would happen in this above case when we pass the default secret location to autopilot?

Reply:
Yes, in that case we don't pass any CA information to autopilot and stork ... see https://github.com/libopenstorage/operator/pull/275/files/528f8c09601e7bb81a43718e2c13cedf18fe5aa1#diff-55d055b011d6ef2d0e392b7906d7716e28b33693712baf32368b3f74e3754597L643
```
		if clusterSpec.Security.TLS != nil && !IsEmptyOrNilCertLocation(clusterSpec.Security.TLS.RootCA) {
			// if no Root CA is specified, we assume server certs are signed by a well known cert authority. Do not need a CA
```
			
I validated this by manually testing as part of https://portworx.atlassian.net/browse/OPERATOR-296

Reply:
Oh right, makes sense!